src/libraries/System.Runtime/tests/System/Type/TypeTests.cs

```diff
@@ -346,23 +329,170 @@ public void MakeArrayType_InvalidArrayRank_ThrowsTypeLoadException(int rank)
         }
 
+        public static IEnumerable<object[]> MakeByRefType_TestData()
+        {
+            return new object[][]
+            {
+                // Primitives
+                new object[] { typeof(string) },
+                new object[] { typeof(sbyte) },
+                new object[] { typeof(byte) },
+                new object[] { typeof(short) },
+                new object[] { typeof(ushort) },
+                new object[] { typeof(int) },
+                new object[] { typeof(uint) },
+                new object[] { typeof(long) },
+                new object[] { typeof(ulong) },
+                new object[] { typeof(char) },
+                new object[] { typeof(bool) },
+                new object[] { typeof(float) },
+                new object[] { typeof(double) },
+                new object[] { typeof(IntPtr) },
+                new object[] { typeof(UIntPtr) },
+                new object[] { typeof(void) },
+
+                // Primitives enums
+                new object[] { typeof(SByteEnum) },
+                new object[] { typeof(ByteEnum) },
+                new object[] { typeof(Int16Enum) },
+                new object[] { typeof(UInt16Enum) },
+                new object[] { typeof(Int32Enum) },
+                new object[] { typeof(UInt32Enum) },
+                new object[] { typeof(Int64Enum) },
+                new object[] { typeof(UInt64Enum) },
+
+                // Array, pointers
+                new object[] { typeof(string[]) },
+                new object[] { typeof(int[]) },
+                new object[] { typeof(int*) },
+
+                // Classes, structs, interfaces, enums
+                new object[] { typeof(NonGenericClass) },
+                new object[] { typeof(GenericClass<int>) },
+                new object[] { typeof(NonGenericStruct) },
+                new object[] { typeof(GenericStruct<int>) },
+                new object[] { typeof(NonGenericInterface) },
+                new object[] { typeof(GenericInterface<int>) },
+                new object[] { typeof(AbstractClass) },
+                new object[] { typeof(StaticClass) },
+
+                // Generic types.
+                new object[] { typeof(GenericClass<>) },
+                new object[] { typeof(GenericClass<>).MakeGenericType(typeof(GenericClass<>)) },
+                new object[] { typeof(GenericClass<>).GetTypeInfo().GetGenericArguments()[0] },
+
+                // By-Ref Like.
+                new object[] { typeof(TypedReference) },
+                new object[] { typeof(ArgIterator) },
+                new object[] { typeof(RuntimeArgumentHandle) },
+                new object[] { typeof(Span<int>) },
+            };
+        }
+
         [Theory]
-        [InlineData(typeof(int))]
+        [MemberData(nameof(MakeByRefType_TestData))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/36642", TestRuntimes.Mono)]
```

Comment:
https://github.com/dotnet/runtime/issues/37489 I guess, but don't reset the PR just for that

Reply:
Fixed in commit